src/test/scala/viper/gobra/GobraTests.scala

```diff
@@ -57,6 +59,12 @@ class GobraTests extends AbstractGobraTests with BeforeAndAfterAll {
 
         val executor: GobraExecutionContext = new DefaultGobraExecutionContext()
         val (result, elapsedMilis) = time(() => Await.result(gobraInstance.verify(config)(executor), Duration.Inf))
+        val startTime = System.currentTimeMillis()
+        // terminate executor with a larger timeout such that we can distinguish a timeout from terminate taking quite long
+        executor.terminate(10 * execution_context_terminate_timeout_ms)
+        val terminateDurationMs = System.currentTimeMillis() - startTime
+        // check whether timeout has been exceeded and fail test accordingly:
+        assert(terminateDurationMs < execution_context_terminate_timeout_ms)
```

Comment:
This code seems to be repeated in various places, could we abstract it in a function?